peas/weather_abstract.py

```diff
@@ -0,0 +1,214 @@
+
+        data['Safe'] = self.safe_dict['Safe']
+        data['Sky condition'] = self.safe_dict['Sky']
+        data['Wind condition'] = self.safe_dict['Wind']
+        data['Gust condition'] = self.safe_dict['Gust']
+        data['Rain condition'] = self.safe_dict['Rain']
+
+        # Store current weather
+        self.weather_entries.append(data)
+
+        if send_message:
+            self.send_message({'data': data}, channel='weather')
+
+        if use_mongo:
+            self.db.insert_current('weather', data)
+
+        return data
+
+    def make_safety_decision(self, current_values):
+        self.logger.debug('Making safety decision')
+        self.logger.debug('Found {} weather data entries in last {:.0f} minutes'.format(
+            len(self.weather_entries), self.safety_delay))
+        safe = False
+
+        # Tuple with condition,safety
+        cloud = self._get_cloud_safety(current_values)
+
+        try:
+            wind, gust = self._get_wind_safety(current_values)
+        except Exception as e:
+            self.logger.warning('Problem getting wind safety: {}'.format(e))
+            wind = ['N/A']
+            gust = ['N/A']
+
+        rain = self._get_rain_safety(current_values)
+
+        safe = cloud[1] & wind[1] & gust[1] & rain[1]
+        self.logger.debug('Weather Safe: {}'.format(safe))
+
+        return {'Safe': safe,
+                'Sky': cloud[0],
+                'Wind': wind[0],
+                'Gust': gust[0],
+                'Rain': rain[0]}
+
+    def _get_cloud_safety(self, sky_diff, last_cloud):
+        safety_delay = self.safety_delay
+
+        # get threshholds for sky-ambient
+        amb_thresholds = self.thresholds['sky-ambient']
+
+        threshold_cloudy = amb_thresholds.get('cloudy')
+        threshold_very_cloudy = amb_thresholds.get('very_cloudy')
+
+        if len(sky_diff) == 0:
+            self.logger.debug('UNSAFE: no sky temperatures found')
+            sky_safe = False
+            cloud_condition = 'Unknown'
+        else:
+            if sky_diff > threshold_very_cloudy[0] and sky_diff <= threshold_very_cloudy[1]:
+                self.logger.debug('UNSAFE: Very cloudy in last {} min. Max sky diff {:.1f} C'.format(
+                                  safety_delay, sky_diff))
+                cloud_condition = 'Very Cloudy'
+                sky_safe = False
+            elif sky_diff > threshold_cloudy[0] and sky_diff <= threshold_cloudy[1]:
+                self.logger.debug('UNSAFE: Cloudy in last {} min. Max sky diff {:.1f} C'.format(
+                                  safety_delay, sky_diff))
+                cloud_condition = 'Cloudy'
+                sky_safe = False
+            else:
+                cloud_condition = 'Clear'
+                sky_safe = True
+
+        self.logger.debug('Cloud Condition: {} (Sky-Amb={:.1f} C)'.format(cloud_condition, last_cloud))
+
+        return cloud_condition, sky_safe
+
+    def _get_wind_safety(self, wind_speed, wind_gust):

```

Comment:
For the sake of modularity and consistency of interface with the other safety decision methods it may be better to split this into separate average wind speed and wind gust methods.

Reply:
Also the wind safety methods should work with the statuses returned from `_get_status()`, not directly with the met data.

Reply:
Modified to use the results from _get_status() and I have split the wind and gust into different methods in the base class.